department-of-veterans-affairs/vets-website
# WIP fix form links get app url calls

## Summary

- Prevents getAppUrl calls from happening on every page load.
- Since the constants were imported for any form and essentially any page in the site, the getAppUrl function would fire for every form id in the FORM_LINKS constant.
- Wrapped the FORM_LINKS constant in a function, memoized the new function, and then only import that memoized function where it is being used, that way it will only ever fire once and use memoized values if it is used several times within an application lifecycle.
- Also added a slightly better error message for if the getAppUrl does happen to get called with an entry name that isn't present in the appRegistry of content-build

## Related issue(s)

- https://github.com/department-of-veterans-affairs/va.gov-team/issues/75760

## Testing done

- Added unit tests for helper do verify memoization.
- Manually tested:
  - To check this out yourself, you can add a console.log to `getAppUrl` in `src/platform/utilities/registry-helpers.js` and load up main, and see that even on the homepage, it would be called a bunch of times.
  - To verify memoization you could load up the MyVa dashboard on main, see how many calls to getAppUrl are happening, then open up this branch and compare. 

## Screenshots

No UI changes

## What areas of the site does it impact?

MyVa, platform level helpers

## Acceptance criteria

- [x] Function calls cleaned up

### Quality Assurance & Testing
